addons/sms/models/sms_sms.py

```diff
@@ -16,74 +18,64 @@ class SmsSms(models.Model):
     _order = 'id DESC'
 
     IAP_TO_SMS_STATE = {
+        'processing': 'process',
         'success': 'sent',
+        'delivered': 'delivered',  # not used for _postprocess_iap_results but webhook events.
         'insufficient_credit': 'sms_credit',
         'wrong_number_format': 'sms_number_format',
         'server_error': 'sms_server',
         'unregistered': 'sms_acc'
     }
 
+    BOUNCE_DLR_ERRORS = {'sms_invalid_destination', 'sms_not_allowed', 'sms_rejected'}
+    DLR_ERRORS = {'sms_expired', 'sms_not_delivered', *BOUNCE_DLR_ERRORS}
+
+    uuid = fields.Char(
+        'UUID', help='UUID used by sms service', copy=False, readonly=True, index=True, default=lambda __: uuid4().hex
+    )
+
     number = fields.Char('Number')
     body = fields.Text()
     partner_id = fields.Many2one('res.partner', 'Customer')
     mail_message_id = fields.Many2one('mail.message', index=True)
     state = fields.Selection([
         ('outgoing', 'In Queue'),
-        ('sent', 'Sent'),
+        ('process', 'Processing'),
+        ('sent', 'Sent'),  # Contrary to notifications and traces, this one does "only" mean sent.
```

Comment:
@nseinlet , @KangOl , would be willing to let us update all "sent" notifications and mailing_traces to a "delivered"  state to avoid the mess above, 
there
https://github.com/odoo/odoo/pull/133392/files#diff-77db957d6e794f1734613afca70ee61352114b05381ab9350fcb5ee1974d9004R88-R89
and 
https://github.com/odoo/odoo/pull/133392/files#diff-f41dc058599fb369e65c3dc160089926be21e1e853de17de516e6cb8bed18eb9R31-R32

(Bonus: while we're re-writing everything, we could also harmonize the other `ready/outgoing`, `cancel/canceled` and `error/exception` for sms, notifications, and traces once and for all).

I suppose that as there are way fewer sms records than notifs/traces we could upgrade `sms_sms` and add `pending` and consider `sent`==`delivered` here too, but can we challenge you to defend that the first (+bonus) option isn't worth it?